bfasst/job.py

```diff
@@ -0,0 +1,34 @@
+"""Job class for a job function and dependency list"""
+
+import uuid
+
+from bfasst.tool import BfasstException
+
+
+class Job:
+    """Class for a job function and dependency list"""
+
+    def __init__(self, function, design_rel_path, dependencies=None):
+        self.function = function
+        self.design_rel_path = design_rel_path
+        self.dependencies = dependencies
+        self.__set_uuid()
+
+    def __set_uuid(self):
```

Comment:
Can you expand this docstring?  What's the purpose of the UUID?  Why not just use the job instance itself as an identifier?

Reply:
We can't use the job instance because the jobs were having problems pickling between processes. A uuid allows us to look up the job in the job list when we need to run a callback (like cleaning up the job list after a job finishes) without having to pickle the job. Hunting down all the pickling errors was a nightmare

Reply:
Sounds good.  Just add a bit of that explanation to the docstring.

Reply:
Done